background-processor/src/lib.rs

```diff
@@ -0,0 +1,240 @@
+		let (temporary_channel_id, tx, funding_output) = match events[0] {
+			Event::FundingGenerationReady { ref temporary_channel_id, ref channel_value_satoshis, ref output_script, user_channel_id } => {
+				assert_eq!(*channel_value_satoshis, channel_value);
+				assert_eq!(user_channel_id, 42);
+
+				let tx = Transaction { version: 1 as i32, lock_time: 0, input: Vec::new(), output: vec![TxOut {
+					value: *channel_value_satoshis, script_pubkey: output_script.clone(),
+				}]};
+				let funding_outpoint = OutPoint { txid: tx.txid(), index: 0 };
+				(*temporary_channel_id, tx, funding_outpoint)
+			},
+			_ => panic!("Unexpected event"),
+		};
+
+		nodes[0].node.funding_transaction_generated(&temporary_channel_id, funding_output);
+		nodes[1].node.handle_funding_created(&nodes[0].node.get_our_node_id(), &get_event_msg!(nodes[0], MessageSendEvent::SendFundingCreated, nodes[1].node.get_our_node_id()));
+		nodes[0].node.handle_funding_signed(&nodes[1].node.get_our_node_id(), &get_event_msg!(nodes[1], MessageSendEvent::SendFundingSigned, nodes[0].node.get_our_node_id()));
+
+		macro_rules! check_persisted_data {
+			($node: expr, $filepath: expr) => {
+				let bytes = loop {
+					match std::fs::read($filepath) {
+						Ok(bytes) => break bytes,
+						Err(_) => continue
+					}
+				};
+				let mut expected_bytes = Vec::new();
+				assert!($node.write(&mut expected_bytes).is_ok());
+				assert_eq!(bytes, expected_bytes);
+				// Check that the ChannelManager reset the persistence lock's bool to
+				// false after signalling for updates.
+				let mutcond = Arc::clone(&$node.persistence_notifier.persistence_lock);
+				let &(ref mtx, _) = &*mutcond;
+				let guard = mtx.lock().unwrap();
+				assert_eq!(*guard, false);
+				mem::drop(guard);
+			}
+		}
+
+		// Check that the initial channel manager data is persisted as expected.
+		let filepath_0 = util::get_full_filepath("persister_0".to_string(), "manager".to_string());
+		let filepath_1 = util::get_full_filepath("persister_1".to_string(), "manager".to_string());
+		check_persisted_data!(nodes[0].node, filepath_0.clone());
+		check_persisted_data!(nodes[1].node, filepath_1.clone());
+
+		// Save the current bytes then trigger another update.
+		let current_bytes = std::fs::read(filepath_0.clone()).unwrap();
+
+		// Force-close the channel.
+		nodes[0].node.force_close_channel(&OutPoint { txid: tx.txid(), index: 0 }.to_channel_id());
+
+		// Loop until what's written on disk changes.
+		loop {
+			match std::fs::read(filepath_0.clone()) {
+				Ok(bytes) => {
+					if bytes == current_bytes { continue } else { break }
+				},
+				Err(_) => continue
+			}
+		};
+		// Make sure the update was persisted properly.
+		check_persisted_data!(nodes[0].node, filepath_0.clone());
+		// TODO(val): `Drop` is never called on the FilesystemPersisters (causing their directories to linger). why?
+		let _ = fs::remove_dir_all("persister_0");
+		let _ = fs::remove_dir_all("persister_1");
+	}
```

Comment:
Logger is available, can you add a failing-persistence test to check that error is logged as expected L54 ?

Reply:
Added `test_persist_error`.